yt/utilities/minimal_representation.py

```diff
@@ -165,64 +148,13 @@ def restore(self, storage, ds):
         pass
 
     def upload(self):
-        api_key = ytcfg.get("yt", "hub_api_key")
-        url = ytcfg.get("yt", "hub_url")
-        if api_key == "":
-            raise YTHubRegisterError
-        metadata, (final_name, chunks) = self._generate_post()
-        if hasattr(self, "_ds_mrep"):
-            self._ds_mrep.upload()
-        for i in metadata:
-            if isinstance(metadata[i], np.ndarray):
-                metadata[i] = metadata[i].tolist()
-            elif hasattr(metadata[i], "dtype"):
-                metadata[i] = np.asscalar(metadata[i])
-        metadata["obj_type"] = self.type
-        if len(chunks) == 0:
-            chunk_info = {"chunks": []}
-        else:
-            chunk_info = {"final_name": final_name, "chunks": []}
-            for cn, cv in chunks:
-                chunk_info["chunks"].append((cn, cv.size * cv.itemsize))
-        metadata = json.dumps(metadata)
-        chunk_info = json.dumps(chunk_info)
-        datagen, headers = multipart_encode(
-            {"metadata": metadata, "chunk_info": chunk_info, "api_key": api_key}
-        )
-        request = urllib.request.Request(url, datagen, headers)
-        # Actually do the request, and get the response
-        try:
-            rv = urllib.request.urlopen(request).read()
-        except urllib.error.HTTPError as ex:
-            if ex.code == 401:
-                mylog.error("You must create an API key before uploading.")
-                mylog.error("https://data.yt-project.org/getting_started.html")
-                return
-            else:
-                raise ex
-        uploader_info = json.loads(rv)
-        new_url = url + "/handler/%s" % uploader_info["handler_uuid"]
-        for i, (cn, cv) in enumerate(chunks):
-            f = TemporaryFile()
-            np.save(f, cv)
-            f.seek(0)
-            pbar = UploaderBar("%s, % 2i/% 2i" % (self.type, i + 1, len(chunks)))
```

Comment:
This is the only place in the code that the UploaderBar class is used. Maybe that should be removed too?

Reply:
nice catch ! Removed it !